File: CS50-Havard-2018/pset6/mario/mario.py
import cs50

while True:
    print("Height: ", end="")
    height = cs50.get_int()
    if height >= 0 and height <= 23:
        break
# print space first, then fill with #s
for i in range(height):
    for j in range(height - i - 1):
        print(" ", end="")
    for k in range(i + 2):
        print("#", end="")
    print("")

File: CS50-Havard-2018/pset1/mario/more/mario.c
# include <stdio.h>
# include <cs50.h>

// This program by Xinhe Liu prints right-aligned pyramid by user input
int main(void)
{
    int height;
    // request user input
    do
    {
        printf("PLease give me a non-negative integer less than 23: ");
        height = get_int();
    }
    while (height < 0 || height > 23);
    // print the pyramid
    int width = height; // the last line has n hashes on the left
    for (int i = 0; i < height; i++)  // i rows
    {
        for (int j = 0; j < width; j++) // j columns
        {
            if (j < width - 1 - i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("  ");
        for (int j = 0; j < i + 1; j++) // j columns
        {
            printf("#");
        }

        printf("\n");
    }
    return 0;
}